solidity/contracts/bridge/WalletProposalValidator.sol

```diff
@@ -587,6 +597,152 @@ contract WalletProposalValidator {
         return true;
     }
 
+    /// @notice View function encapsulating the main rules of a valid moving
+    ///         funds proposal. This function is meant to facilitate the
+    ///         off-chain validation of the incoming proposals. Thanks to it,
+    ///         most of the work can be done using a single readonly contract
+    ///         call.
+    /// @param proposal The moving funds proposal to validate.
+    /// @param walletMainUtxo The main UTXO of the source wallet.
+    /// @return True if the proposal is valid. Reverts otherwise.
+    /// @dev Requirements:
+    ///     - The source wallet must be in the MovingFunds state,
+    ///     - The source wallet must not have any pending redemptions,
+    ///     - The source wallet must not have any pending moved funds sweep
+    ///       requests,
+    ///     - The source wallet's BTC balance must be greater than zero,
+    ///     - The number of target wallets in the proposal must match the
+    ///       expected count, based on the Live wallets count, source wallet's
+    ///       BTC balance and maximum BTC transfer limit,
+    ///     - All target wallets must be in the Live state and ordered correctly
+    ///       (i.e. no duplicates and in ascending order of wallet addresses),
+    ///     - The proposed moving funds transaction fee must be greater than
+    ///       zero,
+    ///     - The proposed moving funds transaction fee must not exceed the
+    ///       maximum total fee allowed for moving funds.
+    function validateMovingFundsProposal(
```

Comment:
This function can be simplified as we can leverage the fact that on-chain commitment submission is also a kind of validation. Looks like it is enough that `validateMovingFundsProposal` makes sure that:
- The source wallet's state is `MovingFunds`
- The source wallet's commitment hash is submitted
- The target wallets from the `proposal ` match the submitted commitment hash (same wallets and same order)
- The proposed fee is greater than zero
- The proposed fee does not exceed the maximum fee

In that model, the coordination leader responsible for generating a moving funds proposal can do the following:
1. Gather all data necessary to assemble a moving funds proposal
2. Submit the commitment (or check if it is already submitted). Before doing that, the leader can make sure the commitment can be submitted using a non-mutating call. This can avoid burning gas if the wallet's state does not allow for commitment submission.
3. Wait for the commitment transaction to be mined
4. Call `validateMovingFundsProposal` to make sure the proposal is actually valid and can be safely broadcasted
5. Return the proposal

On the other hand, coordination followers receiving a moving funds proposal should:
1. Call `validateMovingFundsProposal` to make sure the proposal is actually submitted and valid
2. Wait several blocks to confirm proposal validity (32 blocks?). This is needed to avoid corner cases where the commitment is submitted but not included in the canonical chain due to a reorg and another front-running transaction that affects commitment validity.
3. Call `validateMovingFundsProposal` again to ensure the commitment has not been changed during the idleness period
4. Assemble, sign, and broadcast moving funds transaction on Bitcoin

Reply:
Done in 3f7d6b3687df77a8402605f8575024dc95f2a33c.

Reply:
Leaving it open for future reference!